src/pybind/mgr/ansible_orchestrator/module.py

```diff
@@ -0,0 +1,364 @@
+        {'name': 'server_addr'},
+        {'name': 'server_port'},
+        {'name': 'username'},
+        {'name': 'password'},
+        {'name': 'certificate'}  # Ansible runner https server certificate file
+    ]
+
+    def __init__(self, *args, **kwargs):
+        """
+        """
+        super(Module, self).__init__(*args, **kwargs)
+
+        self.run = False
+
+        self.all_completions = []
+
+        self.ar_client = None
+
+    def available(self):
+        """ Check if Ansible Runner service is working
+        """
+        # TODO
+        return (True, "Everything ready")
+
+    def wait(self, completions):
+        """Given a list of Completion instances, progress any which are
+           incomplete.
+
+           @param completions: list of Completion instances
+           @Returns          : List with completions operations pending
+        """
+
+        # Check progress and update status in each operation
+        for operation in completions:
+            self.log.info("playbook <%s> status:%s", operation.playbook, operation.status)
+            if operation.is_complete:
+                operation.update_result()
+
+        completions = filter(lambda x: not x.is_complete, completions)
+
+        self.log.info("Operations pending: %s", len(completions))
+
+        return completions
+
+    def serve(self):
+        """ Mandatory for standby modules
+        """
+        self.log.info("Starting Ansible Orchestrator module ...")
+
+        # Verify config options (just that we have data)
+        self.verify_config()
+
+        # Ansible runner service client
+        self.ar_client = Client(server = self.get_config('server_addr', ''),
+                                port = self.get_config('server_port', ''),
+                                user = self.get_config('username', ''),
+                                password = self.get_config('password', ''),
+                                certificate = self.get_config('certificate', ''),
+                                logger = self.log)
+
+        if not self.ar_client.is_operative():
+            self.log.error("Ansible Runner Service not available. "
+                          "Check external server status or connection options. "
+                          "If configuration options changed try to "
+                          "disable/enable the module.")
+            self.shutdown()
+            return
+
+
+        self.run = True
+
+    def shutdown(self):
+        self.log.info('Stopping Ansible orchestrator module')
+        self.run = False
+
+    def get_inventory(self, node_filter=None):
+        """
+
+        @param   :	node_filter instance
+        @Return  :	A AnsibleReadOperation instance (Completion Object)
+        """
+
+        # Create a new read completion object for execute the playbook
+        ansible_operation = AnsibleReadOperation(client = self.ar_client,
+                                                 playbook = GET_INVENTORY_PLAYBOOK,
```

Comment:
`playbook` is confusing if we are actually called an inventory

Reply:
I should leave the name of the constant without changes, although i will add the following comment over the definition of the constant.
Name of the playbook used in the "get_inventory" method. This playbook is expected to provide a list of storage devices in the host where the playbook is executed.

Reply:
done